src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,139 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private Object[] elementData;
+    private int size = 0;
+    private static final int DEFAULT_LIST_CAPACITY = 10;
+
+    public ArrayList() {
+        elementData = new Object[]{};
+    }
+
+    public ArrayList(int length) {
+        if (length > 0) {
+            elementData = new Object[length];
+        } else if (length == 0) {
+            elementData = new Object[]{};
+        } else {
+            throw new ArrayIndexOutOfBoundsException("Your index is lesser that 0");
+        }
+    }
 
     @Override
     public void add(T value) {
-
+        if (elementData.length == 0 || size == elementData.length) {
+            grow();
+            add(value);
```

Comment:
```suggestion
            elementData[size++] = value;
```
aaand now you don't need else block :)